src/librustc_borrowck/borrowck/mir/dataflow/mod.rs

```diff
@@ -0,0 +1,1168 @@
+            repr::TerminatorKind::Call { ref cleanup, ref destination, func: _, args: _ } => {
+                if let Some(ref unwind) = *cleanup {
+                    self.propagate_bits_into_entry_set_for(in_out, changed, unwind);
+                }
+                if let Some((ref dest_lval, ref dest_bb)) = *destination {
+                    // N.B.: This must be done *last*, after all other
+                    // propagation, as documented in comment above.
+                    self.flow_state.operator.propagate_call_return(
+                        &self.ctxt, in_out, bb, *dest_bb, dest_lval);
+                    self.propagate_bits_into_entry_set_for(in_out, changed, dest_bb);
+                }
+            }
+        }
+    }
+
+    fn propagate_bits_into_entry_set_for(&mut self,
+                                         in_out: &[usize],
+                                         changed: &mut bool,
+                                         bb: &repr::BasicBlock) {
+        let entry_set = self.flow_state.sets.for_block(bb.index()).on_entry;
+        let set_changed = bitwise(entry_set, in_out, &self.flow_state.operator);
+        if set_changed {
+            *changed = true;
+        }
+    }
+}
+
+// Dataflow analyses are built upon some interpretation of the
+// bitvectors attached to each basic block, represented via a
+// zero-sized structure.
+//
+// Note on PhantomData: Each interpretation will need to instantiate
+// the `Bit` and `Ctxt` associated types, and in this case, those
+// associated types need an associated lifetime `'tcx`. The
+// interpretive structures are zero-sized, so they all need to carry a
+// `PhantomData` representing how the structures relate to the `'tcx`
+// lifetime.
+//
+// But, since all of the uses of `'tcx` are solely via instances of
+// `Ctxt` that are passed into the `BitDenotation` methods, we can
+// consistently use a `PhantomData` that is just a function over a
+// `&Ctxt` (== `&MoveData<'tcx>).
+
+/// `MaybeInitializedLvals` tracks all l-values that might be
+/// initialized upon reaching a particular point in the control flow
+/// for a function.
+///
+/// For example, in code like the following, we have corresponding
+/// dataflow information shown in the right-hand comments.
+///
+/// ```rust
+/// struct S;
+/// fn foo(pred: bool) {                       // maybe-init:
+///                                            // {}
+///     let a = S; let b = S; let c; let d;    // {a, b}
+///
+///     if pred {
+///         drop(a);                           // {   b}
+///         b = S;                             // {   b}
+///
+///     } else {
+///         drop(b);                           // {a}
+///         d = S;                             // {a,       d}
+///
+///     }                                      // {a, b,    d}
+///
+///     c = S;                                 // {a, b, c, d}
+/// }
+/// ```
+///
+/// To determine whether an l-value *must* be initialized at a
```

Comment:
The complement of `MaybeUninitialized` (i.e. `DefinitelyInitialized`) is enough.